commands/group_test.go

```diff
@@ -166,3 +166,303 @@ func (s *MmctlUnitTestSuite) TestTeamGroupEnableCmd() {
+			GetGroupsByChannel(channelPart, *groupOpts).
+			Return(nil, 0, &model.Response{Error: &mockError}).
+			Times(1)
+
+		err := channelGroupEnableCmdF(s.client, &cobra.Command{}, []string{channelArg})
+		s.Require().NotNil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.EqualError(err, mockError.Error())
+	})
+
+	s.Run("Enable group constrains with PatchChannel error", func() {
+		printer.Clean()
+
+		teamArg := "team-id"
+		mockTeam := model.Team{Id: teamArg}
+		channelPart := "channel-id"
+		mockChannel := model.Channel{Id: channelPart}
+		channelArg := teamArg + ":" + channelPart
+		group := &model.Group{Name: "group-name"}
+		mockGroups := []*model.Group{group}
+		mockError := model.AppError{Id: "Mock Error"}
+		groupOpts := &model.GroupSearchOpts{
+			PageOpts: &model.PageOpts{
+				Page:    0,
+				PerPage: 10,
+			},
+		}
+
+		s.client.
+			EXPECT().
+			GetTeam(teamArg, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetChannelByNameIncludeDeleted(channelPart, teamArg, "").
+			Return(&mockChannel, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetGroupsByChannel(channelPart, *groupOpts).
+			Return(mockGroups, 0, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			PatchChannel(channelPart, &model.ChannelPatch{GroupConstrained: model.NewBool(true)}).
+			Return(nil, &model.Response{Error: &mockError}).
+			Times(1)
+
+		err := channelGroupEnableCmdF(s.client, &cobra.Command{}, []string{channelArg})
+		s.Require().NotNil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.EqualError(err, mockError.Error())
+	})
+
+	s.Run("Enable group constrains with no associated groups", func() {
+		printer.Clean()
+
+		teamArg := "team-id"
+		mockTeam := model.Team{Id: teamArg}
+		channelPart := "channel-id"
+		mockChannel := model.Channel{Id: channelPart}
+		channelArg := teamArg + ":" + channelPart
+		mockGroups := []*model.Group{}
+		groupOpts := &model.GroupSearchOpts{
+			PageOpts: &model.PageOpts{
+				Page:    0,
+				PerPage: 10,
+			},
+		}
+
+		s.client.
+			EXPECT().
+			GetTeam(teamArg, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetChannelByNameIncludeDeleted(channelPart, teamArg, "").
+			Return(&mockChannel, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetGroupsByChannel(channelPart, *groupOpts).
+			Return(mockGroups, 0, &model.Response{Error: nil}).
+			Times(1)
+
+		err := channelGroupEnableCmdF(s.client, &cobra.Command{}, []string{channelArg})
+		s.Require().NotNil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.EqualError(err, "Channel '"+channelArg+"' has no groups associated. It cannot be group-constrained")
+	})
+
+	s.Run("Enable group constrains with nonexistant team", func() {
```

Comment:
Small typo here and in the test case below, should be `nonexistent`.

Reply:
Thank you for the feedback. This has been corrected now.